researchapp/sync_api.py

```diff
@@ -0,0 +1,62 @@
+''' Library for interfacing with the "synchronizer".
+'''
+import requests
+
+
+class SyncExtension(object):
+    ''' Flask extension for interacting with the Synchronizer API.
+    '''
+    def __init__(self, app=None):
+        self.session = None
+        self.host = None
+
+        if app:
+            self.init_app(app)
+
+    def init_app(self, app):
+        ''' Init the extension.
+        '''
+        self.session = requests.Session()
+        self.host = app.config['SYNCHRONIZER_HOST']
+
+    def list_providers(self):
+        ''' List available providers.
+        '''
+        resp = self.session.get(self.host + '/providers')
+        return resp.json()
+
+    def create_participant(self):
+        ''' Create a new participant.
+        '''
+        resp = self.session.post(self.host + '/participants')
+        return resp.json()
+
+    def create_authorization(self, participant_id, provider_id, redirect_uri):
+        ''' Store an authorization.
+        '''
+        path = '/participants/{}/authorizations/{}'.format(participant_id, provider_id)
+        params = {
```

Comment:
`redirect_uri` actually contains the URI. The expected format of the redirect uri is `http://example.com/authorized?code=blablabla&state=blablabla`.